### PR TITLE
FogL 2182 - safe mode startup option

### PR DESCRIPTION
**Pull request by ashish-jabble**
**Fixed items**

- [x] foglamp start --safe-mode added
- [x] text added when run in safe mode for `foglamp start and foglamp status`
- [x] safeMode added in /ping, related ticket for GUI https://scaledb.atlassian.net/browse/FOGL-2216
- [x] Changes required for Scheduler code as well
- [x] Audit message updated for both modes on start/stop
- [x]  Restart should be normal mode

### CODE REVIEW

File: python/foglamp/services/core/server.py [hunk 1]
```diff
@@ -582,8 +585,10 @@ def _check_readings_table(cls, loop):
 
     @classmethod
     def _start_core(cls, loop=None):
-        _logger.info("start core")
+        if cls.running_in_safe_mode:
```
Comment (amarendra-dianomic) [inline python/foglamp/services/core/server.py]: An explanatory comment will be better to explain what behaviour is expected from core running in safe-mode.

File: scripts/foglamp [hunk 2]
```diff
@@ -22,7 +22,7 @@ set -e
 #
 # This is the startup script for foglamp
 #
-USAGE="Usage: `basename ${0}` {start|stop|status|reset|kill|help|version}"
+USAGE="Usage: `basename ${0}` {start|start --safe-mode|stop|status|reset|kill|help|version}"
```
Comment (amarendra-dianomic) [inline scripts/foglamp]: Seems that only "--" is prefixed to safe-mode only.

Comment (ashish-jabble) [reply scripts/foglamp]: JIRA has asked for `--safe_mode` and I only replaced underscore with hyphen i.e `--safe-mode `

Comment (praveen-garg) [reply scripts/foglamp]: @amarendra-dianomic  Yeah. As it is a  flag with start, hence `start --safe-mode`

File: python/foglamp/services/core/server.py [hunk 3]
```diff
@@ -582,8 +585,10 @@ def _check_readings_table(cls, loop):
...
+            _logger.info("Starting in safe mode")
 
+        _logger.info("start core")
```
Comment (praveen-garg) [inline python/foglamp/services/core/server.py]: Let's  log  once. `_logger.info("Starting ...") and _logger.info("Starting in SAFE MODE...")` based on condition

**Comment by praveen-garg** [comment]
In safe mode, I see syslog as:

```
Dec 20 17:12:15 FogLAMP[10098] INFO: server: foglamp.services.core.server: start scheduler
Dec 20 17:12:15 FogLAMP[10098] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Starting Scheduler: Management port received is 33715
Dec 20 17:12:15 FogLAMP[10098] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'purge' to start at 2018-12-20 18:12:15.390403
Dec 20 17:12:15 FogLAMP[10098] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'stats collection' to start at 2018-12-20 17:12:30.390403
Dec 20 17:12:15  FogLAMP[10098] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Scheduled task for schedule 'certificate checker' to start at ...
```

We should not!

the next log statement to 
```
Dec 20 17:12:15 FogLAMP[10098] INFO: server: foglamp.services.core.server: start scheduler
```
should be

```
In safe mode, the scheduler will be in restricted mode. Only the Admin API operations and current state of schedules will be accessible; No jobs / processes will be triggered.
```

**Pull request merged**